pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -9,6 +9,28 @@ shopt -s inherit_errexit
 
 export PATH=@path@:$PATH
 
+
+# save output piped to msgPipe or 'msg message' to systemd journal
+# 'journalctl -t nixos-rebuild' to inspect
+# tee is used so we still get output to the terminal
+# check if systemd-cat exists and if systemctl exits with 0. if it does not
+# then systemd isn't running(chroot) and we cannot use systemd-cat.
+
+# msg something
+msg() {
+  if command -v systemd-cat &>/dev/null && systemctl &>/dev/null; then
+    echo -e "$@" | tee >(systemd-cat -t nixos-rebuild)
```

Comment:
Is `-e` a good idea here? What if some arbitrary command output contains any backslashes? `echo` is also problematic since it accepts flags and even accepts *different* flags based on the shell. For example, if the first argument is `-n`:

```
$ messages=('-n' 'foo') && echo "${messages[@]}" | xxd
00000000: 666f 6f                                  foo
```

Basically, you might want to [use `printf '%s\n' "$@"` instead](https://unix.stackexchange.com/q/65803/3645).

Reply:
I think the above printf should use "$*" (single arg) instead of "$@" (multiple args).

Reply:
Why? It's almost always a bad idea to use `$*`, as explained in for example [What is the difference between $* and $@?](https://unix.stackexchange.com/q/41571/3645) and [`for arg in $*`](https://mywiki.wooledge.org/BashPitfalls#for_arg_in_.24.2A).

Reply:
I'd have to test it, but I thought the printf format string had to match the number of arguments?

Reply:
`printf` will reuse the format string repeatedly as long as there are arguments to process, for example:

```
$ printf '%s\n' 'foo bar' 'baz'
foo bar
baz
```

Conversely, if there are too *few* arguments it will fill in an empty string, for example:

```
$ printf '%s-%s-%s\n' 'foo bar' 'baz'
foo bar-baz-
```

Reply:
Thanks for the info, TIL.